elastic/apm-server
# Add new http.response fields for transactions and spans

## Motivation/summary


This PR adds new fields to the Intake API for `spans` and `transactions`. The fields were requested from the RUM agent developers, but it was agreed that they should also be available for backend agents. 
Following fields are added to also support  navigation and resource timing for http responses:
```
context.http.response.transfer_size (optional, number)
context.http.response.encoded_body_size (optional, number)
context.http.response.decoded_body_size (optional, number
```

Additionally `context.http.response.headers` are added for `span` events. 

Since this PR introduces the `context.http.response` group for spans, the currently used `context.http.status_code` is deprecated with this PR and instead the field is moved inside the `response` group for alignment.

## Checklist




## How to test these changes


Start APM Server and send a request with 
* spans and transactions containing the newly introduced fields
* spans and transactions not containing the newly introduced fields, as they are optional
* spans and transactions containing the newly introduced field keys but with `nil` values, as they are optional
* spans containing the deprecated field `context.http.status_code` and ensure it is stored
* spans containing the deprecated field `context.http.status_code` and new field `context.http.response.status_code` and ensure the new field overwrites the old one

## Related issues
Closes #3234
